handlers/upload.go

```diff
@@ -193,3 +200,26 @@ func (d *CatalystAPIHandlersCollection) processUploadVOD(streamName, sourceURL,
 
 	return nil
 }
+
+func createDtsh(requestID, destination string) error {
+	log.Log(requestID, "Generating DTSH header for ", destination)
+	url, err := url.Parse(destination)
+	if err != nil {
+		return err
+	}
+	url.RawQuery = ""
+	url.Fragment = ""
+	headerPrepare := exec.Command(path.Join(config.PathMistDir, "MistInMP4"), "-H", url.String(), "-g", "5")
+	if err = subprocess.LogOutputs(headerPrepare); err != nil {
+		return err
+	}
+	if err = headerPrepare.Start(); err != nil {
+		return err
+	}
+	go func() {
+		if err := headerPrepare.Wait(); err != nil {
```

Comment:
Why do we do the `Wait()` in an async function? Don't we need to wait for it to finish to get the benefits of the DTSH file being there?

Reply:
Good catch! My mistake from time when dtsh was being created after transcode response

Reply:
Yup, good catch - missed this as well.